### ISSUE TITLE
[CR] Defining Dynamic Game Events in JSON

### ISSUE THREAD

**Issue by vache** [CleverRaven/Cataclysm-DDA#7943]
So, first thing's first: I'm probably not going to actually do this, I just want to plant the seed in the mind of someone who might be willing to.

The idea is to create a system supporting the definition of generic events in json, and a trigger system to set them off.  The end goal of this system is to allow a more dynamic "alive" game world.

There are a few components that I can see a system like this needing:
A set of effects.  This can be anything from "create an explosion", "create a loud noise", "display a game message", "summon a monster", etc.
A set of shapes, or ways of defining an location to be effected.  Basically, what area will the effect affect.  
A set of trigger conditions.  This is what happens to cause the effect.  A few I had in mind would be: "activated from a computer", "enter a certain tile/zone", "passed a noise threshold", "picked up a certain item", "killed a certain monster", etc.
A set of behavioral definitions (possibly).  Depending on the trigger, what specifics are behind the trigger?  Does it trigger only once, or every time the trigger gets tripped?

An example would be if you are somewhere with a nuclear reactor.  You activate the control terminal and withdraw the control rods (trigger).  This causes the temperature and radiation to rise (effect) around the reactor (area).  When the radiation and temperature hit a certain level (trigger), an explosion (effect) triggers and an alarm (effect) sounds, while safety doors close (effect + location).

This is sort of like what I had in mind for my computer refactor, but then I got to thinking about traps today, and how the same ideas could be applied to trap behavior to create scripted-style "set-pieces" for certain zones.  The big thing is that it could all be json definable so they can go into user created maps without having to compile in custom functions.
##

**Comment by illi-kun** [comment]
### This issue was closed as it appears inactive.

Reducing open issues to those which are (or will) be actively worked upon helps us focus our efforts. This issue has not been deleted - it still appears in searches and if it contains relevant information you are encouraged to continue to link to it.

### If this issue was a bug
It should be reopened if it can be reproduced in the current build. You can obtain the most recent copy [here](http://dev.narc.ro/cataclysm/jenkins-latest/). Please check there is not a more recent report of this bug before doing so. If no more recent report exists you should continue the discussion in this issue.

### If this was a feature request
If the consensus was that the idea was good you could consider submitting an implementation via a PR. If you want to comment further please do so here as opposed to opening a new issue. Before posting check nobody has already made the same point and consider whether your comments are likely to lead to an implementation. If you have doubts about either consider instead [voting](http://iv.cataclysmdda.com:8111/all) for the issue

### If you want to work on this issue
Then either assign it to yourself or if you are unable to do so claim it via adding a comment. Please don't assign others or make a general request for action.
